cmd/cloud-controller-manager/highperfcontroller.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+	cloudprovider "k8s.io/cloud-provider"
+	networkclientset "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned"
+	networkinformer "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/network/v1"
+	alphanetworkinformer "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/network/v1alpha1"
+	highperfcontroller "k8s.io/cloud-provider-gcp/pkg/controller/highperf"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	"k8s.io/cloud-provider/app"
+	cloudcontrollerconfig "k8s.io/cloud-provider/app/config"
+	genericcontrollermanager "k8s.io/controller-manager/app"
+	"k8s.io/controller-manager/controller"
+)
+
+func startHighPerfControllerWrapper(initCtx app.ControllerInitContext, config *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) app.InitFunc {
+	return func(ctx context.Context, controllerCtx genericcontrollermanager.ControllerContext) (controller.Interface, bool, error) {
+		return startHighPerfController(config, controllerCtx, cloud)
+	}
+}
+
+func startHighPerfController(ccmConfig *cloudcontrollerconfig.CompletedConfig, controllerCtx genericcontrollermanager.ControllerContext, cloud cloudprovider.Interface) (controller.Interface, bool, error) {
+	gceCloud, ok := cloud.(*gce.Cloud)
+	if !ok {
+		err := fmt.Errorf("HighPerfController does not support %v provider", cloud.ProviderName())
+		return nil, false, err
+	}
+
+	kubeConfig := ccmConfig.Complete().Kubeconfig
+	kubeConfig.ContentType = "application/json" //required to serialize HighPerf objects to json
+	networkClient, err := networkclientset.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, false, err
+	}
+
+	nwInformer := networkinformer.NewNetworkInformer(networkClient, 0*time.Second, cache.Indexers{})
```

Comment:
note for myself, if we keep adding informers for the same types we should start thinking in consolidating and pass ithem as parameter